spec/features/recipes/show_recipe_page_spec.rb

```diff
@@ -0,0 +1,21 @@
+feature 'user visits show recipe page' do
+  before do
+    FactoryBot.create(:recipe, :scrambled_eggs)
+    FactoryBot.create(:recipe, :tikka_masala)
+    visit('/')
+    page.all(:xpath, '//a[contains(text(), "Show")]')[1].click
+  end
+
+  it 'shows user clicked recipe' do
+    expect(page).to have_current_path(%r{recipes/\d+})
+
+    expect(page).to have_css("//img[src*='.png']")
```

Comment:
This is too general, please narrow it (to name of the file?) because now it will evaluate to `true` for each `png` file on the page.

Reply:
Sorry, but I cannot find any information about how to inject regexp into xpath matcher so I have to leave it for now.